### PR TITLE
connectors: streamline invocations and end-to-end tests

### PR DESCRIPTION
**Pull request by jgraettinger**
**Description:**

Internal refactors to simplify connector connector invocations for captures and materializations.

I've also attempted to simplify end-to-end tests. There's still an occasional flake I've seen in CI runs which has been very fickle to track down. I've left in a bit of debug logging to help narrow down what's happening should it re-occur.

No user-facing changes to behavior, etc.

**Workflow steps:**

No changes.

**Documentation links affected:**

No changes.

**Notes for reviewers:**

Recommend commit-by-commit.

---
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/565)

### CODE REVIEW

File: go/connector/run.go [hunk 1]
```diff
@@ -96,120 +109,28 @@ func Run(
...
-			jsonFiles[imageInspectJsonFileName] = inspectOutput
-		}
-	}
-
-	args = append([]string{
-		fmt.Sprintf("--image-inspect-json-path=/tmp/%s", imageInspectJsonFileName),
-		fmt.Sprintf("--log.level=%s", ops.LogrusToFlowLevel(logger.Level()).String()),
+		"--log-driver", "none",
+		// Network to which the container should attach.
+		"--network", networkName,
```
Comment (psFried) [inline go/connector/run.go:114]: I don't think this will work in the case that `networkName` is empty. `--network=` might work OK though?

Comment (jgraettinger) [reply go/connector/run.go:120]: Updated to use `bridge` if networkName is empty. I think this is compatible with your PR as well. Good?

**Review by mdibaiee — APPROVED** [review]
LGTM except a note regarding `:local`

File: go/connector/run.go [hunk 2]
```diff
@@ -565,38 +400,40 @@ func (fe *firstError) unwrap() error {
...
-
-func PullRemoteImage(ctx context.Context, image string, logger ops.Logger) error {
-	var combinedOutput, err = exec.CommandContext(ctx, "docker", "pull", image).CombinedOutput()
-	logger.Log(logrus.TraceLevel, nil, fmt.Sprintf("output from docker pull: %s", combinedOutput))
-	if err != nil {
-		return fmt.Errorf("pull image: %w", err)
+// PullImage to local cache unless the tag is `:local-test-tag`, which is expected to be local.
+func PullImage(ctx context.Context, image string) error {
+	// Pull the image if it's not expected to be local.
+	if strings.HasSuffix(image, ":local-test-tag") {
```
[the remark this answers is not in the document]
Comment (jgraettinger) [reply go/connector/run.go:406]: Yep!

File: go/connector/run.go (hunk 1 above)
Comment (psFried) [reply go/connector/run.go:120]: Sounds good to me :+1:

**Pull request merged**